js/views/modals/Settings/Settings.js

```diff
@@ -0,0 +1,87 @@
+import loadTemplate from '../../../utils/loadTemplate';
+import BaseModal from '../BaseModal';
+import SettingsGeneral from './SettingsGeneral';
+import SettingsPage from './SettingsPage';
+import $ from 'jquery';
+
+export default class extends BaseModal {
+  constructor(options = {}) {
+    const opts = {
+      removeOnClose: true,
+      modelContentClass: 'modalContent clrP border clrBr',
+      ...options,
+    };
+
+    super(opts);
+    this.options = opts;
+
+    this.currentTabView = '';
+    this.TabViews = { SettingsGeneral, SettingsPage };
+
+    if (this.options.removeOnClose) this.on('close', () => this.remove());
+  }
+
+  className() {
+    return `${super.className()} settings`;
+  }
+
+  events() {
+    return {
+      'click .js-tab': 'tabClick',
+      'click .js-cancel': 'cancelClick',
+      'click .js-save': 'saveClick',
+      ...super.events(),
+    };
+  }
+
+  tabClick(e) {
+    const targ = $(e.target).closest('.js-tab');
+    this.selectTab(targ);
+  }
+
+  selectTab(targ) {
+    const targTab = targ.data('tab');
+    const TabView = `Settings${targTab}`;
+    let newTabView;
+
+    this.$('.js-tab').removeClass('clrT active');
+    targ.addClass('clrT active');
+
+    if (this.TabViews[TabView]) {
```

Comment:
Is this if check really necessary?

If the developer forgot to add a child view class into `this.TabViews` or if they put the wrong name in the html data attribute, wouldn't it be better to just let the compiler throw an exception on the next line (`this.TabViews[TabView]` will bomb)? Instead, what will happen now is that the target for the new tab will be active, but the contents will still be from the previous tab.